src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt

```diff
@@ -963,15 +963,17 @@ class LicenceService(
     return Pair(firstName, lastName)
   }
 
-  private fun findOriginalLicenceForVariation(variationLicence: VariationLicence): CrdLicence {
+  private fun findOriginalLicenceForVariation(variationLicence: VariationLicence): EntityLicence {
     var originalLicence = variationLicence
     while (originalLicence.variationOfId != null) {
       val licence = licenceRepository
         .findById(originalLicence.variationOfId!!)
         .orElseThrow { EntityNotFoundException("${originalLicence.variationOfId}") }
       when (licence) {
         is CrdLicence -> return licence
+        is HardStopLicence -> return licence
         is VariationLicence -> originalLicence = licence
+        else -> error("Unknown licence type in hierarchy: ${licence.javaClass}")
```

Comment:
Loudly fail if other types are introduced (can't use sealed classes here due to using inheritance)